ci-operator/step-registry/openshift/e2e/gcp/libvirt/cert-rotation/conf/openshift-e2e-gcp-libvirt-cert-rotation-conf-commands.sh

```diff
@@ -0,0 +1,439 @@
+}
+
+jumpstartNodes () {
+  approveCSRs
+  # jumpstart any stuck nodes, during recovery nodes will be rebooted
+  nodesDisabled=$(oc get nodes | grep "NotReady" | awk '{ print $1 }')
+  if [ ! -z "${nodesDisabled}" ]; then
+    nodeDisabledList=( $nodesDisabled )
+    for i in "${nodeDisabledList[@]}"
+    do
+      echo "Restarting stuck node ${i}..."
+      sudo virsh destroy "${i}"
+      sleep 30
+      sudo virsh start "${i}"
+      sleep 60
+    done
+    checkNodesReady
+  fi
+}
+
+checkDegradedCOs () {
+  retries=0
+  # image-pruner job in openshift-image-registry namespace may be stuck due to time skew. This would not
+  # happen if time was progressing naturally. Kill image-prune jobs here.
+  oc delete jobs --all -n openshift-image-registry
+  sleep 10
+  while ! oc wait co --all --for='condition=Degraded=False' --timeout=10s && [ $retries -lt 50 ]; do
+    (( retries++ ))
+  done
+}
+
+checkProgressingCOs () {
+  retries=0
+  # image-pruner job in openshift-image-registry namespace may be stuck due to time skew. This would not
+  # happen if time was progressing naturally. Kill image-prune jobs here.
+  oc delete jobs --all -n openshift-image-registry
+  sleep 10
+  while ! oc wait co --all --for='condition=Progressing=False' --timeout=10s && [ $retries -lt 50 ]; do
+    jumpstartNodes
+    (( retries++ ))
+  done
+}
+
+checkAvailableCOs () {
+  retries=0
+  while ! oc wait co --all --for='condition=Available=True' --timeout=20s && [ $retries -lt 100 ]; do
+    jumpstartNodes
+    (( retries++ ))
+  done
+}
+
+sudo systemctl stop chronyd
+
+SKEW=${1:-+1y}
+
+OC=${OC:-oc}
+SSH=${SSH:-ssh}
+
+masters=$( ${OC} get nodes --selector='node-role.kubernetes.io/master' --template='{{ range $index, $_ := .items }}{{ range .status.addresses }}{{ if (eq .type "InternalIP") }}{{ if $index }} {{end }}{{ .address }}{{ end }}{{ end }}{{ end }}' )
+workers=$( ${OC} get nodes --selector='!node-role.kubernetes.io/master' --template='{{ range $index, $_ := .items }}{{ range .status.addresses }}{{ if (eq .type "InternalIP") }}{{ if $index }} {{end }}{{ .address }}{{ end }}{{ end }}{{ end }}' )
+
+function run-on {
+        for n in ${1}; do ${SSH} core@"${n}" sudo 'bash -eEuxo pipefail' <<< ${2}; done
+}
+
+ssh-keyscan -H ${masters} ${workers} >> ~/.ssh/known_hosts
+
+run-on "${masters} ${workers}" "systemctl stop kubelet"
+
+# Destroy all containers on workers.
+run-on "${workers}" "crictl rm --all -f"
+# Destroy all containers on masters except KAS and etcd.
+run-on "${masters}" '
+kas_id=$( crictl ps --name="^kube-apiserver$" -q )
+[[ -n "${kas_id}" ]]
+etcd_id=$( crictl ps --name="^etcd$" -q )
+[[ -n "${etcd_id}" ]]
+other_ids=$( crictl ps --all -q | ( grep -v -e "${kas_id}" -e "${etcd_id}" || true ) )
+if [ -n "${other_ids}" ]; then
+        crictl rm -f ${other_ids}
+fi;
+'
+
+# Delete all pods, especialy the operators. Makes sure it needs KCM and KS working when starting again.
+${OC} delete pods -A --all --force --grace-period=0 --timeout=0
+
```

Comment:
```suggestion

# Delete all clusteroperator status to avoid stale status when the operator pod isn't started.
export bearer=$( ioc -n openshift-cluster-version serviceaccounts get-token default ) && export server=$( oc whoami --show-server ) && for co in $( ioc get co --template='{{ range .items }}{{ printf "%s\n" .metadata.name }}{{ end }}' ); do curl -X PATCH -H "Authorization: Bearer ${bearer}" -H "Accept: application/json" -H "Content-Type: application/merge-patch+json" ${server}/apis/config.openshift.io/v1/clusteroperators/${co}/status -d '{"status": null}' -k && echo; done
```

Reply:
added

Reply:
4.7 run passed, 4.6 run failed, with the above addition so that's good